httpbakery/client_test.go

```diff
@@ -219,21 +275,17 @@ func (r *readCounter) Read(buf []byte) (int, error) {
 	return n, err
 }
 
-func newService(c *gc.C, location string, locator bakery.PublicKeyLocator) *bakery.Service {
+func newService(location string, locator bakery.PublicKeyLocator) *bakery.Service {
 	svc, err := bakery.NewService(bakery.NewServiceParams{
 		Location: location,
 		Locator:  locator,
 	})
-	c.Assert(err, gc.IsNil)
+	if err != nil {
```

Comment:
I tend to use panic the logic is extremely unlikely to fail and if it does that's nothing to
do with what's being tested.

In this case, NewService can fail only because the system random number
generator returns an error, which will never happen on any of the usual
platforms AFAIK.